torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -660,6 +667,11 @@ class NormalizationScheduler : public SchedulerEntry {
         return false;
       }
     }
+
+    if (!scheduler_utils::registerPersistentBufferCheck(fusion, runtime_info)) {
```

Comment:
When we re-evaluate heuristics for new sizes do we always call canSchedule? I thought this should be caught in heuristics meaning this should be called in heuristics. Probably if we fail this in heuristics we should return an empty heuristic

Reply:
`When we re-evaluate heuristics for new sizes do we always call canSchedule?`
Yes because the segments now are input dependent. When evaluating new sizes, the kernel cache first checks `canSchedule` still true before getting the actual schedule params.

The use model for `canSchedule` and `getHeuristics` are basically  `canSchedule` defines if a scheduler can generate correct kernel on a segmented group or complete fusion, and `getHeuristics` goes into computing the actual parameters assuming `canSchedule` has already been checked.

The case where persistence is required but there's no enough persistent buffer seems to be a `canSchedule` logic to me so I just moved it here and segmenter also needs to be sure not to produce segments like that.